File: rumireaZvyagintsevaLesson4/cryptoloader/src/main/java/com/example/cryptoloader/MainActivity.java
package com.example.cryptoloader;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.loader.app.LoaderManager;
import androidx.loader.content.Loader;

import com.example.cryptoloader.databinding.ActivityMainBinding;

import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.InvalidParameterException;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;

import javax.crypto.BadPaddingException;
import javax.crypto.Cipher;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.KeyGenerator;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;

public class MainActivity extends AppCompatActivity implements LoaderManager.LoaderCallbacks<String> {

    private ActivityMainBinding binding;
    private static final int LOADER_ID = 1234;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.buttonEncrypt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String phrase = binding.editTextPhrase.getText().toString();

                if (phrase.isEmpty()) {
                    Toast.makeText(MainActivity.this, "введите фразу", Toast.LENGTH_SHORT).show();
                    return;
                }

                SecretKey key = generateKey();
                byte[] cipherText = encryptMsg(phrase, key);

                Bundle bundle = new Bundle();
                bundle.putByteArray(MyLoader.ARG_WORD, cipherText);
                bundle.putByteArray("key", key.getEncoded());

                LoaderManager.getInstance(MainActivity.this).restartLoader(LOADER_ID, bundle, MainActivity.this);
            }
        });
    }

    @NonNull
    @Override
    public Loader<String> onCreateLoader(int id, @Nullable Bundle args) {
        if (id == LOADER_ID) {
            Toast.makeText(this, "onCreateLoader: " + id, Toast.LENGTH_SHORT).show();
            return new MyLoader(this, args);
        }
        throw new InvalidParameterException("Invalid loader id");
    }

    @Override
    public void onLoadFinished(@NonNull Loader<String> loader, String data) {
        if (loader.getId() == LOADER_ID) {
            Toast.makeText(this, "дешифрованная фраза: " + data, Toast.LENGTH_LONG).show();
        }
    }

    @Override
    public void onLoaderReset(@NonNull Loader<String> loader) {
    }

    public static SecretKey generateKey() {
        try {
            SecureRandom sr = SecureRandom.getInstance("SHA1PRNG");
            sr.setSeed("any data used as random seed".getBytes(StandardCharsets.UTF_8));
            KeyGenerator kg = KeyGenerator.getInstance("AES");
            kg.init(256, sr);
            return new SecretKeySpec(kg.generateKey().getEncoded(), "AES");
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    public static byte[] encryptMsg(String message, SecretKey secret) {
        try {
            Cipher cipher = Cipher.getInstance("AES");
            cipher.init(Cipher.ENCRYPT_MODE, secret);
            return cipher.doFinal(message.getBytes(StandardCharsets.UTF_8));
        } catch (NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException |
                 BadPaddingException | IllegalBlockSizeException e) {
            throw new RuntimeException(e);
        }
    }

    public static String decryptMsg(byte[] cipherText, SecretKey secret) {
        try {
            Cipher cipher = Cipher.getInstance("AES");
            cipher.init(Cipher.DECRYPT_MODE, secret);
            return new String(cipher.doFinal(cipherText), StandardCharsets.UTF_8);
        } catch (NoSuchAlgorithmException | NoSuchPaddingException |
                 IllegalBlockSizeException | BadPaddingException | InvalidKeyException e) {
            throw new RuntimeException(e);
        }
    }
}

File: rumireaZvyagintsevaLesson4/app/src/main/java/com/example/rumireazvyagintsevalesson4/MainActivity.kt
package com.example.rumireazvyagintsevalesson4

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.rumireazvyagintsevalesson4.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var isPlaying = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.textViewSong.text = "Blinding Lights"
        binding.textViewArtist.text = "The Weeknd"
        binding.textViewStatus.text = "Статус: остановлено"
        binding.buttonPlay.text = "Play"

        binding.buttonPlay.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                isPlaying = !isPlaying

                if (isPlaying) {
                    binding.buttonPlay.text = "Pause"
                    binding.textViewStatus.text = "Статус: воспроизведение"
                } else {
                    binding.buttonPlay.text = "Play"
                    binding.textViewStatus.text = "Статус: пауза"
                }
            }
        })

        binding.buttonNext.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                binding.textViewSong.text = "Save Your Tears"
                binding.textViewArtist.text = "The Weeknd"
                binding.textViewStatus.text = "Статус: следующий трек"
                isPlaying = false
                binding.buttonPlay.text = "Play"
            }
        })

        binding.buttonPrev.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                binding.textViewSong.text = "Starboy"
                binding.textViewArtist.text = "The Weeknd"
                binding.textViewStatus.text = "Статус: предыдущий трек"
                isPlaying = false
                binding.buttonPlay.text = "Play"
            }
        })
    }
}